### PR TITLE
Initial notebook to generate omega scans from background files

### PR DESCRIPTION
**Pull request by ricco-hub**
Made notebook that generates omega scans for the top _n_ highest detection statistics from the `background.h5` files for both Livingston and Hanford. Saves plots as `png` files.

### CODE REVIEW

**Comment by wbenoit26** [comment]
This looks nice! Can you convert it to a python file so I can leave comments on the relevant lines? Just remove the `%matplotlib inline`, add a function:
```
def main(n: int, background_file: str, background_dir: str):
    times, shifts = find_max_times(background_file, n)
    load_background(background_dir, times=times, shifts=shifts)
```

and put the following at the bottom of the script:
```
if __name__ == "__main__":
    background_file = '/home/ricco.venterea/results/my-first-run/infer/background.h5'
    background_dir = '/home/ricco.venterea/data/test/background'
    n = 2
    main(background_file, background_dir, n)
```

And everything should work the same as before.

File: projects/plots/plots/omega_scan.py [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                plot_L1.show()
+                plot_L1.savefig('omega_scan_L1_{time}_{number}.png'.format(time=t, number=counter))
+            else:
+                continue
+
+def main(n: int, background_file: str, background_dir: str):
+    times, shifts = find_max_times(background_file, n)    
+    load_background(background_dir, times=times, shifts=shifts)
+
+if __name__ == "__main__":
```
Comment (wbenoit26) [inline projects/plots/plots/omega_scan.py:83]: We'll want to set this up so we can run it as part of the `luigi` structure that Ethan has set up, but I'll add comments about that on a future pass through. For now, let's address these comments and make sure everything works.

**Review by wbenoit26 — CHANGES_REQUESTED** [review]

**Comment by EthanMarx** [comment]
This is awesome, will be nice to have this automated.

File: projects/plots/plots/omega_scan.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+def find_max_times(path:str, n:int=1) -> np.ndarray:
+    '''
+    path, path to background.h5 file
+    n, n highest maxima (e.g. n=2, top 2 highest detection statistic)
+
+    return central times times for n highest detection statistics given background path
+    '''
+        
+    background = h5py.File(path,'r')
```
[the remark this answers is not in the document]
Comment (EthanMarx) [reply projects/plots/plots/omega_scan.py:17]: This obviously works, but would make some things cleaner if you used the `EventSet` object directly from the ledger. i.e.

`background = EventSet.read(path)`

File: projects/plots/plots/omega_scan.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+        start_dur = b.index('-', start+1)
+        period = b.index('.')
+        end_dur = b.index('.', period+1)
+        duration = b[start_dur+1:end_dur]
+
+        end_time = start_time + duration
+
+        for counter, t in enumerate(times):
+            if int(start_time) <= t <= int(end_time):
```
[the remark this answers is not in the document]
Comment (ricco-hub) [reply projects/plots/plots/omega_scan.py:56]: I don't recall having an issue but just wanted to make sure this inequality remained true; I'll remove it.

**Pull request closed without merging**